src/base/bittorrent/torrenthandle.cpp

```diff
@@ -433,17 +434,28 @@ QHash<QString, TrackerInfo> TorrentHandle::trackerInfos() const
 
 void TorrentHandle::addTrackers(const QVector<TrackerEntry> &trackers)
 {
-    QSet<TrackerEntry> currentTrackers;
-    for (const lt::announce_entry &entry : m_nativeHandle.trackers())
-        currentTrackers << entry;
-
     QVector<TrackerEntry> newTrackers;
     newTrackers.reserve(trackers.size());
 
+    uint8_t nextTier = 0;
+    const std::vector<lt::announce_entry> currentTrackers = m_nativeHandle.trackers();
```

Comment:
This should be on its own line for readability and `newTrackers` is too far away from its usage... I'll just give you what I had in mind:
```c++
{
    const std::vector<lt::announce_entry> currentTrackers = m_nativeHandle.trackers();

    int nextTier = 0;
    if (!currentTrackers.empty()) {
        for (const lt::announce_entry &entry : currentTrackers)
            nextTier = std::max<int>(nextTier, entry.tier);
        ++nextTier;
        nextTier = std::min<int>(nextTier, std::numeric_limits<uint8_t>::max());
    }

    QSet<TrackerEntry> currentTrackersSet;
    for (const lt::announce_entry &entry : currentTrackers)
        currentTrackersSet << entry;

    QVector<TrackerEntry> newTrackers;
    newTrackers.reserve(trackers.size());

    for (const TrackerEntry &tracker : trackers) {
        if (!currentTrackersSet.contains(tracker)) {
            TrackerEntry trackerToAdd {tracker};
            trackerToAdd.setTier(nextTier);
            m_nativeHandle.add_tracker(trackerToAdd.nativeEntry());
            newTrackers << trackerToAdd;
        }
    }

    if (!newTrackers.isEmpty())
        m_session->handleTorrentTrackersAdded(this, newTrackers);
}
```